v3/src/models/data/data-set-types.ts

```diff
@@ -58,6 +58,10 @@ export interface IMoveAttributeCollectionOptions extends IMoveAttributeOptions {
   collection?: string // id of destination collection; undefined => no collection (ungrouped)
 }
 
+export interface IAttributeChangeResult {
+  removedCollection?: boolean // True if changing the attribute resulted in its collection being removed
```

Comment:
How about making this a string and returning the id of the removed collection (if any)?